DEV_GUIDE.md

```diff
@@ -0,0 +1,152 @@
+# PlayerUI Typescript Development Guide
+
+### Sections:
+1. [Setup](#Setup)
+2. [Adding a new plugin](#adding-a-new-plugin)
+
+## Setup
+### Demo Application
+The demo app can be built and launched from the command line with bazel:
+```bash
+bazel build //docs/site:site
+bazel run //docs/site:site
+```
+
+## Adding a new plugin
+Adding a new plugin is simple using predefined macros for `BUILD.bazel` files.
+
+### Core vs React Plugins
+Core plugins provide shared functionality, usable on all platforms.
+
+React plugins add react specific functionality.
+
+### Scaffold
+All plugins follow a common structure:
+
+The basic structure is as follows:
+```bash
+plugins/example
+├── react
+│   ├── BUILD.bazel
+│   ├── src
+│   │   ├── index.tsx
+│   │   └── __tests__
+│   │       └── index.test.tsx
+│   └── package.json
+└── core
+    ├── BUILD.bazel
+    ├── src
+    │   ├── index.ts
+    │   └── __tests__
+    │       └── index.test.ts
+    └── package.json
+```
+
+To scaffold your plugin, create a new create a new `core` or `react` folder in the appropriate plugin directory, with the appropriate folders, a package.json and a blank `BUILD.bazel`:
+
+For core plugins:
+```bash
+mkdir -p plugins/example-plugin/core/src
+cd plugins/example-plugin/core
+pnpm init
+touch BUILD.bazel
+```
+
+For react plugins:
+```bash
+mkdir -p plugins/example-plugin/react/src
+cd plugins/example-plugin/react
+pnpm init
+touch BUILD.bazel
+```
+
+Set the name, version, and entry point of the package in the `package.json`:
+
+For core plugins:
+```javascript
+{
+  "name": "@player-ui/example-plugin",
+  "version": "0.0.0-PLACEHOLDER",
+  "main": "src/index.ts",
+}
+```
+
+For react plugins:
+```javascript
+{
+  "name": "@player-ui/example-plugin",
+  "version": "0.0.0-PLACEHOLDER",
+  "main": "src/index.tsx",
+}
+```
+
+### BUILD
+The `js_pipeline` macro will handle the full build:
+
+```python
+load("@rules_player//javascript:defs.bzl", "js_pipeline")
+
+js_pipeline(package_name = "@player-ui/example-plugin")
+```
+
+In order to use the `tsup` bundler, add `tsup_config` to the `BUILD.bazel` file. This is done by loading the `tsup_config` helper from the `tools` package and calling it within `BUILD.bazel`:
+
+```python
+load("//tools:defs.bzl", "tsup_config")
+
+tsup_config(name = "tsup_config")
+```
+
```

Comment:
I would maybe move the info on the vitest setup here to colocate all the `BUILD` file setup in one place